pkg/registry/core/service/ipallocator/controller/repair.go

```diff
@@ -156,13 +156,16 @@ func (c *Repair) doRunOnce() error {
 	snapshotByFamily := make(map[v1.IPFamily]*api.RangeAllocation)
 	storedByFamily := make(map[v1.IPFamily]ipallocator.Interface)
 
-	err := wait.PollImmediate(time.Second, 10*time.Second, func() (bool, error) {
+	var lastError error
+	var snapshot *api.RangeAllocation
+	err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 30*time.Second, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
please add a comment explaining why 30 seconds

Reply:
as @xuzhenglun mentioned before, other controllers has use 30s > 20s. the 30s is enough for controller timeout 1min.

Reply:
the timeout in the outside is 60s, would it be better if make it align with the outside? Because if a timeout occurs here and the timeout is less than 60 seconds, we still need to wait for the outer layer's 60 seconds to trigger the Fatal

Reply:
this fails apiserver to start IIRC, the first doRunOnce is required to succeed